BUILDING.md

```diff
@@ -0,0 +1,68 @@
+## Build the development environment
+
+The first prerequisite of properly building SwarmKit targets is to have a Go
+development environment setup. Please follow [How to Write Go Code](https://golang.org/doc/code.html)
+for proper setup. If done correctly, you should have a GOROOT and GOPATH set in the
+environment.
+
+If a Go development environment is setup, one can use `go get` to install the
+`swarmkit`:
+
+    go get github.com/docker/swarmkit/...
+
+The above will install the source repository into the `GOPATH`. Any binaries
+should be built into `$GOPATH/bin`. Alternatively, use the above with `-d` to
+just download the source.
+
+> __NOTE:__ While you do not need to use `go get` to checkout the SwarmKit
+> project, for these build instructions to work, the project must be checked
+> out in the correct location in the `GOPATH`. This should almost always be
+> `$GOPATH/src/github.com/docker/swarmkit`.
+
+### Repeatable Builds
+
+For the full development experience, one should `cd` into
+`$GOPATH/src/github.com/docker/swarmkit. From there, the regular `go`
+commands, such as `go test`, should work per package (please see
+[Developing](#developing) if they don't work).
+
+A `Makefile` has been provided as a convenience to support repeatable builds.
+The following will install tools into `GOPATH` for use with developing
```

Comment:
Docker provides a `Makefile` as a convenience to support repeatable builds. `make setup` installs tools onto the `GOPATH` for use with developing `SwarmKit`: